packages/terra-application-layout/src/menu/_ApplicationMenu.jsx

```diff
@@ -0,0 +1,179 @@
+};
+
+class ApplicationMenu extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.renderHeader = this.renderHeader.bind(this);
+    this.renderExtensions = this.renderExtensions.bind(this);
+    this.renderFooter = this.renderFooter.bind(this);
+    this.handleUtilityDiscloseRequest = this.handleUtilityDiscloseRequest.bind(this);
+    this.handleUtilityOnChange = this.handleUtilityOnChange.bind(this);
+  }
+
+  handleUtilityDiscloseRequest(utilityMenu) {
+    const { app } = this.props;
+
+    if (app && utilityMenu) {
+      app.disclose({
+        preferredType: modalDisclosureType,
+        size: availableDisclosureSizes.SMALL,
+        content: {
+          component: <UtilityMenuWrapper>{utilityMenu}</UtilityMenuWrapper>,
+          key: 'application-menu-utility-menu',
+        },
+      });
+    }
+  }
+
+  handleUtilityOnChange(event, key) {
+    const { utilityConfig, app } = this.props;
+
+    utilityConfig.onChange(event, key, app && app.disclose);
+  }
+
+  renderHeader(isCompact) {
+    const { nameConfig } = this.props;
+
+    if (isCompact && (nameConfig.accessory || nameConfig.title)) {
+      return (
+        <div className={cx(['menu-header'])}>
+          <ApplicationMenuName accessory={nameConfig.accessory} title={nameConfig.title} />
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  renderExtensions(isCompact) {
+    const { app, layoutConfig, extensions } = this.props;
+
+    if (isCompact && extensions) {
+      return React.cloneElement(extensions, { app, layoutConfig });
+    }
+
+    return null;
+  }
+
+  renderFooter(isCompact) {
+    const { utilityConfig } = this.props;
+
+    if (isCompact && utilityConfig) {
+      return (
+        <ApplicationMenuUtility
+          onChange={this.handleUtilityOnChange}
+          onDisclose={this.handleUtilityDiscloseRequest}
+          title={utilityConfig.title}
+          accessory={utilityConfig.accessory}
+          menuItems={utilityConfig.menuItems}
+          initialSelectedKey={utilityConfig.initialSelectedKey}
+          data-application-menu-utility
+        />
+      );
+    }
+
+    return null;
+  }
+
+  render() {
+    const {
+      app,
+      content,
+      extensions,
+      layoutConfig,
+      nameConfig,
+      routingStackDelegate,
+      navigationLayoutRoutes,
+      navigationLayoutSize,
+      utilityConfig,
+      ...customProps
+    } = this.props;
+
+    const menuClassNames = cx([
+      'application-menu',
+      customProps.className,
+    ]);
+
+    const isCompact = Helpers.isSizeCompact(layoutConfig.size);
+
+    let clonedContent;
+    if (content) {
+      clonedContent = React.cloneElement(content, { app, layoutConfig, routingStackDelegate, navigationLayoutRoutes, navigationLayoutSize });
+    }
+
+    return (
+      <div {...customProps} className={menuClassNames}>
+        <ApplicationMenuLayout
+          header={this.renderHeader(isCompact)}
```

Comment:
It would improve readability if we check `isCompact` before rending the header, extensions or footer since each of these helpers return null if `!isCompact`